lib/min_heap.rb

```diff
@@ -14,18 +14,24 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(nlogn) because heap_up is nlogn, where n is the number of nodes in the heap
+  # Space Complexity: O(1)
```

Comment:
Since `heap_up` is recursive, you have to account for the space complexity of the system stack.